arch/x86/core/ia32/swap.S

```diff
@@ -363,6 +369,17 @@ CROHandlingDone:
 	pushl arch_timing_swap_start
 	popl arch_timing_value_swap_temp
 time_read_not_needed:
+#endif
+#if defined(CONFIG_TRACING)
+
+	/* Save these as we are using to keep track of isr and isr_param */
+	pushl	%eax
+	pushl	%edx
```

Comment:
we do need to preserve `arch_swap()`'s return value in `%eax`, but why are we pushing/popping `%edx`

Also editorial complaint that we have two different instrumentation points right after one another. What's the difference between `CONFIG_EXECUTION_BENCHMARKING` and `CONFIG_TRACING`? Can we unify?

Reply:
> Also editorial complaint that we have two different instrumentation points right after one another. What's the difference between `CONFIG_EXECUTION_BENCHMARKING` and `CONFIG_TRACING`? Can we unify? Maybe we can do this in a follow-up PR but it sticks out

not mentioned anywhere here but the idea is to drop CONFIG_EXECUTION_BENCHMARKING completely, been working with Daniel on this already and showing progress. This is just the first step toward this